cmd/handler-api.go

```diff
@@ -54,23 +54,31 @@ func (t *apiConfig) init(cfg api.Config, setDriveCounts []int) {
 
 	var apiRequestsMaxPerNode int
 	if cfg.RequestsMax <= 0 {
-		stats, err := sys.GetStats()
+		var maxMem uint64
+		memStats, err := mem.VirtualMemory()
 		if err != nil {
-			logger.LogIf(GlobalContext, err)
 			// Default to 8 GiB, not critical.
-			stats.TotalRAM = 8 << 30
+			maxMem = 8 << 30
+		} else {
+			// MinIO will use only upto 90% of available memory.
+			maxMem = (memStats.Available * 90) / 100
```

Comment:
This doesn't really account for the 50% overallocation of the GC.

```suggestion
			maxMem = memStats.Available / 2
```